eng/pipelines/templates/stages/archetype-java-release-batch.yml

```diff
@@ -0,0 +1,502 @@
+                displayName: 'Verify version for release of ${{ artifact.name }}'
+
+              - pwsh: |
+                  $(Build.SourcesDirectory)/eng/versioning/scan_for_unreleased_dependencies.ps1 ${{artifact.groupId}} ${{artifact.name}} $(Build.SourcesDirectory)/sdk/${{parameters.ServiceDirectory}}
+                displayName: "Verify no unreleased dependencies for ${{ artifact.name }}"
+                # Uber JARs release a flattened POM, as dependencies are incorporated into the JAR circumnavigating the
+                # need to list dependencies in the POM. The flattened POM is generated and won't comply with our
+                # versioning validation, so skip it.
+                condition: and(succeeded(), ne('${{ artifact.uberJar }}', 'true'))
+
+              - ${{if and(ne(parameters.SDKType, 'data'), ne(artifact.skipVerifyChangelog, 'true'))}}:
+                - template: /eng/common/pipelines/templates/steps/verify-changelog.yml
+                  parameters:
+                    PackageName: ${{artifact.name}}
+                    ServiceName: ${{parameters.ServiceDirectory}}
+                    ForRelease: true
+            # Given a list of libraries to release, verify that the list of libraries has full transitive closure.
+            # This prevents us from releasing a library whose dependency is not being released.
+            # Note: Where-Object -Not "uberJar" skips verifiction of uberJars which would fail during creation. It's
+            #       also worth mentioning that because uberJar is on the Where-Object it also needs to be one of the
+            #       objects selected in Select-Object
+            - task: PowerShell@2
+              displayName: Verify Release Set
+              inputs:
+                pwsh: true
+                filePath: $(Build.SourcesDirectory)/eng/versioning/verify_release_set.ps1
+                arguments: >
+                  -ServiceDirectory $(Build.SourcesDirectory)/sdk/${{parameters.ServiceDirectory}}
+                  -ArtifactsList ('${{ convertToJson(parameters.Artifacts) }}' | ConvertFrom-Json | Select-Object name, groupId, uberJar | Where-Object -Not "uberJar")
+
+        - deployment: TagRepository
+          displayName: "Create release tag"
+          condition: and(succeeded(), ne(variables['Skip.TagRepository'], 'true'))
+          environment: github
+          dependsOn: VerifyReleaseVersion
+          variables:
+            - template: /eng/pipelines/templates/variables/globals.yml
+          pool:
+            name: azsdk-pool-mms-win-2019-general
+            vmImage: windows-2019
+          strategy:
+            runOnce:
+              deploy:
+                steps:
+                  - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+                  - template: /eng/common/pipelines/templates/steps/retain-run.yml
+                  - ${{ each artifact in parameters.Artifacts }}:
```

Comment:
Similarly we should eliminate this template loop and just download all the artifacts at one go and try to use the `'${{ convertToJson(parameters.Artifacts) }}'` pattern inline instead.

Reply:
I did this for the download here. create-tags-and-git-release still requires the loop.